Source/Scene/Points3DTileContent.js

```diff
@@ -164,159 +197,394 @@ define([
+            });
+        }
+
+        var normalsVertexBuffer;
+        if (hasNormals) {
+            normalsVertexBuffer = Buffer.createVertexBuffer({
+                context : context,
+                typedArray : normals,
+                usage : BufferUsage.STATIC_DRAW
+            });
+        }
+
+        var positionAttributeLocation = 0;
+        var colorAttributeLocation = 1;
+        var normalAttributeLocation = 2;
+
+        var attributes = [];
+        if (isQuantized) {
+            attributes.push({
+                index : positionAttributeLocation,
+                vertexBuffer : positionsVertexBuffer,
+                componentsPerAttribute : 3,
+                componentDatatype : ComponentDatatype.UNSIGNED_SHORT,
+                normalize : true, // Convert position to 0 to 1 before entering the shader
+                offsetInBytes : 0,
+                strideInBytes : 0
+            });
         } else {
-            fs += '    gl_FragColor = highlightColor; \n';
+            attributes.push({
+                index : positionAttributeLocation,
+                vertexBuffer : positionsVertexBuffer,
+                componentsPerAttribute : 3,
+                componentDatatype : ComponentDatatype.FLOAT,
+                normalize : false,
+                offsetInBytes : 0,
+                strideInBytes : 0
+            });
         }
 
-        fs += '} \n';
+        if (hasColors) {
+            var colorComponentsPerAttribute = isTranslucent ? 4 : 3;
+            attributes.push({
+                index : colorAttributeLocation,
+                vertexBuffer : colorsVertexBuffer,
+                componentsPerAttribute : colorComponentsPerAttribute,
+                componentDatatype : ComponentDatatype.UNSIGNED_BYTE,
+                normalize : true,
+                offsetInBytes : 0,
+                strideInBytes : 0
+            });
+        }
 
-        var boundingSphere = this._tile.contentBoundingVolume.boundingSphere;
+        if (hasNormals) {
+            if (isOctEncoded16P) {
+                attributes.push({
+                    index : normalAttributeLocation,
+                    vertexBuffer : normalsVertexBuffer,
+                    componentsPerAttribute : 2,
+                    componentDatatype : ComponentDatatype.UNSIGNED_BYTE,
+                    normalize : false,
+                    offsetInBytes : 0,
+                    strideInBytes : 0
+                });
+            } else {
+                attributes.push({
+                    index : normalAttributeLocation,
+                    vertexBuffer : normalsVertexBuffer,
+                    componentsPerAttribute : 3,
+                    componentDatatype : ComponentDatatype.FLOAT,
+                    normalize : false,
+                    offsetInBytes : 0,
+                    strideInBytes : 0
+                });
+            }
+        }
 
-        // TODO: performance test with 'interleave : true'
```

Comment:
I think this is still a validate `PERFORMANCE_TODO`.  Regardless of the best payload format over the wire, we also need to put some time into the best runtime format.  Generally, interleaved is only very slightly better so it is not worth it, but I don't know if that has changed in recent years or is different here given the high vertex load and simple shaders.